packages/hash/frontend/src/lib/subgraph.ts

```diff
@@ -386,3 +386,29 @@ export const getEntityTypesByBaseUri = (
     return vertex.inner;
   });
 };
+
+// ------------------- Checked Subgraph Objects -------------------
+
+export type SingleEntityRootedSubgraph = Subgraph & { root: Entity };
+
+/**
+ * Checks if the `subgraph` is rooted at a single `Entity` and modifies the object to store a reference to it as
+ * a `root` property
+ * @param subgraph
+ * @throws if there were more or less than one root
+ * @throws if the root wasn't an `EntityVertex`
+ */
+export const isSingleEntityRootedSubgraph = (
+  subgraph: Subgraph,
+): subgraph is SingleEntityRootedSubgraph => {
+  if (subgraph.roots.length !== 1) {
+    throw new Error(
```

Comment:
> Should I just return false here because this is technically a type guard?

Type guards shouldn't error if the input is not of the type, but still a valid input.

Reply:
Gotcha, this has been made redundant anyway: https://github.com/hashintel/hash/pull/1242/commits/d91204a4e95303ce6d890f86438267532fe3761c